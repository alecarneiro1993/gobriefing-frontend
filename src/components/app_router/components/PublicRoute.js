import React from "react";
import { Route, Redirect } from "react-router";
import PropTypes from "prop-types";

import { HOME_PATH } from "../helpers";

export function PublicRoute({
  isAuthenticated,
  isRestricted,
  component: Component,
  ...rest
}) {
  return (
    <Route
      {...rest}
      render={props =>
        isAuthenticated && isRestricted ? (
          <Redirect to={HOME_PATH} />
        ) : (
          <Component {...props} />
        )
      }
    />
  );
}

PublicRoute.propTypes = {
  path: PropTypes.string.isRequired,
  exact: PropTypes.bool.isRequired,
  component: PropTypes.func.isRequired,
  isAuthenticated: PropTypes.bool,
  isRestricted: PropTypes.bool.isRequired
};

PublicRoute.defaultProps = {
  isAuthenticated: false
};
