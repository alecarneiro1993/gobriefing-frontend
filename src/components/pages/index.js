export { default as SignInPage } from './SignInPage';
export { default as RootPage } from './RootPage';
export { default as HomePage } from './HomePage';
