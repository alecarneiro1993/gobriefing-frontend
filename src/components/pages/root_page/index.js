import React from "react";

class RootPage extends React.Component {
  render() {
    return <div>Root page</div>;
  }
}

export { RootPage };
