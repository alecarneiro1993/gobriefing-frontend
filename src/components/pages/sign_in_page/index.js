import React from "react";

class SignInPage extends React.Component {
  render() {
    return <div>Login</div>;
  }
}

export { SignInPage };
