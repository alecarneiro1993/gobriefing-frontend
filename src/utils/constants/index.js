export * from './routes';
export * from './pages';
export * from './values';
export * from './types';
