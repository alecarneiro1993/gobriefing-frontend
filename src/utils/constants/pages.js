export const ROOT_PAGE = "RootPage";
export const SIGN_IN_PAGE = "SignInPage";
export const HOME_PAGE = "HomePage";
