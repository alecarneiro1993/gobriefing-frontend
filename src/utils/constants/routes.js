export const ROOT_PATH = "/";
export const SIGN_IN_PATH = "/sign_in";
export const HOME_PATH = "/home";
