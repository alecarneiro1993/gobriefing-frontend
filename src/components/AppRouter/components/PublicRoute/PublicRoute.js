// @flow

import React from 'react';
import { Route, Redirect } from 'react-router';

import { HOME_PATH } from 'utils/constants';

type Props = {
  isAuthenticated: boolean,
  isRestricted: boolean,
  component: Class<any>
};

function PublicRoute(props: Props) {
  const {
    isAuthenticated,
    isRestricted,
    component: Component,
    ...rest
  } = props;
  return (
    <Route
      {...rest}
      render={props =>
        isAuthenticated && isRestricted ? (
          <Redirect to={HOME_PATH} />
        ) : (
          <Component {...props} />
        )
      }
    />
  );
}

export default PublicRoute;
