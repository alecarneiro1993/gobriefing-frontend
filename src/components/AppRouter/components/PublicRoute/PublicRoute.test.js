import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router';
import { Route } from 'react-router-dom';

import { ROOT_PATH, SIGN_IN_PATH, HOME_PATH } from 'utils/constants/routes';
import { ROOT_PAGE, SIGN_IN_PAGE } from 'utils/constants/pages';

import PublicRoute from './PublicRoute';

describe('PublicRoute component', () => {
  const Page = (page) => function PageComponent() {
    return <div>{page}</div>;
  };

  it('renders without crashing', () => {
    const props = {
      path: ROOT_PATH,
      component: Page(ROOT_PAGE),
      exact: true,
      isAuthenticated: false,
      isRestricted: false,
    };
    const wrapper = shallow(<PublicRoute {...props} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Route)).toHaveLength(1);
  });

  describe('when its not restricted', () => {
    it('renders the desired component', () => {
      const props = {
        path: ROOT_PATH,
        component: Page(ROOT_PAGE),
        exact: true,
        isAuthenticated: false,
        isRestricted: false,
      };
      const wrapper = mount(
        <MemoryRouter initialEntries={['/']}>
          <PublicRoute {...props} />
        </MemoryRouter>,
      );
      expect(wrapper.find('Router').props().history.location.pathname).toEqual(
        ROOT_PATH,
      );
    });
  });

  describe('when its restricted', () => {
    it('redirects to the home authenticated page', () => {
      const props = {
        path: SIGN_IN_PATH,
        component: Page(SIGN_IN_PAGE),
        exact: true,
        isAuthenticated: true,
        isRestricted: true,
      };
      const wrapper = mount(
        <MemoryRouter initialEntries={['/sign_in']}>
          <PublicRoute {...props} />
        </MemoryRouter>,
      );
      expect(wrapper.find('Router').props().history.location.pathname).toEqual(
        HOME_PATH,
      );
    });
  });
});
