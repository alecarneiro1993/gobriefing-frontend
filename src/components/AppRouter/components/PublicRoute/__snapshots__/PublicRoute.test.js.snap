// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicRoute component renders without crashing 1`] = `
<Route
  exact={true}
  path="/"
  render={[Function]}
/>
`;
